indico/modules/events/registration/badges.py

```diff
@@ -134,8 +134,9 @@ def _draw_badge(self, canvas, person, template, tpl_data, pos_x, pos_y):
                 as_list=True
             ):
                 item_data.update(update)
-            self._draw_item(canvas, item_data['item'], tpl_data, item_data['text'], item_data['pos_x'],
-                            item_data['pos_y'])
+            if item_data['text'] is not None:
```

Comment:
FYI this is what my initial comment about "what if render() returns None" was about:

![image](https://github.com/indico/indico/assets/179599/49f9726f-a2d3-46d7-9f9f-7e6bc661313b)

Before adding this fix, the badge got a literal `None` printed on it.